runtime/syntax/testdir/README.txt

```diff
@@ -26,14 +26,33 @@ Creating a syntax plugin test
 -----------------------------
 
 Create a source file in the language you want to test in the "input"
-directory.  Make sure to include some interesting constructs with complicated
-highlighting.
-
-Use the filetype name as the base and a file name extension matching the
-filetype.  Let's use Java as an example.  The file would then be
+directory.  Use the filetype name as the base and a file name extension
+matching the filetype.  Let's use Java as an example. The file would then be
 "input/java.java".
 
+Make sure to include some interesting constructs with plenty of complicated
+highlighting.  Optionally, pre-configure the testing environment for this file
+by writing in it a series of comments within the top 20 lines and marking just
+as many such lines with an uppercase word "TEST_SETUP" that must be followed
+by a valid Vim Ex command, extending to the end of line.  Continuing a Java
+example:
+
+	// TEST_SETUP let g:java_space_errors = 1
+	// TEST_SETUP let g:java_minlines = 5
+	class Test { }
+
+Remember that no heuristic is set in place to discover whether the matched
+lines belong to a comment or any other valid or invalid construct of the
+language under test; therefore, only for Vim configuration can "TEST_SETUP"
+appear within the top 20 lines.  As an alternative, you can keep the same Vim
```

Comment:
Would you rather prefer this variant?

> Remember that no heuristic is set in place to discover whether the matched
> lines belong to a comment or any other valid or invalid construct of the
> language under test; therefore, only the first 20 lines of the input file are
> scanned for Vim TEST_SETUP commands.

Comments or not comments, any `TEST_SETUP` lines in the top  
20 lines of a test file are pre-empted on behalf of Vim  
configuration.  And we need to be honest about this fact.  
So that test authors would exercise care by not having  
within those lines `TEST_SETUP` labels in `switch-case`  
statements, `TEST_SETUP` macros, etc.

Can we rack our brains and come up with a better wording?

Reply:
Would changing this section:

> The first 20 lines of the input file are scanned for setup commands and these
> will be executed before the syntax highlighting is enabled. Typically, these
> lines would be included as comments so as not to introduce any syntax errors in
> the input file but this is not required.

to this variant read better:

> The first 20 lines of the input file are always scanned for setup commands and these
> will be executed before the syntax highlighting is enabled.  Typically, these
> lines would be included as comments so as not to introduce any syntax errors in
> the input file but this is not required. However, every TEST_SETUP line is
> treated as Vim configuration in the first 20 lines.  Please do not confuse Vim
> by naming TEST_SETUP your case labels in switch statements, your macros, etc.

and the whole sentence of "Remember that no heuristic etc."  
be expunged?

Reply:
Okay, then ALWAYS in caps may suffice?

> The first 20 lines of the input file are ALWAYS scanned for setup commands and
> these will be executed before the syntax highlighting is enabled.  Typically,
> these lines would be included as comments so as not to introduce any syntax
> errors in the input file but this is not required.
>
> Continuing the Java example:

And no "Remember that no heuristic etc." sentence.